integrations/fluid.py

```diff
@@ -20,19 +20,17 @@ def __init__(self):
             None,
             None,
         )
+        self.blocknumber_to_susdeVaults = {}
 
     def get_balance(self, user: str, block: int) -> float:
         balance = 0
-        try:
-            userPositions, vaultEntireDatas = call_with_retry(vaultResolver_contract.functions.positionsByUser(user), block)
-            for userPosition in userPositions:
-                if vaultEntireDatas[0][3][8][0] != susde:
-                    continue
-                balance += userPosition[9]
-            return balance/1e18
-        except Exception as e:
-            return 0
-
+        relevant_vaults = self.get_relevant_vaults(block)
```

Comment:
on this I think the previous implementation must have been more performant 

because a end user will most likely only have < 5 positions, maybe less. so before we iterated through ~5 positions, now we iterate through all user positions of every sUSDE vault?

Reply:
I had mentioned using getAllVaultPositions for get participants, not for get_balance?

Reply:
reverted back to older version now😅